modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb

```diff
@@ -0,0 +1,129 @@
+      #
+      # @param power_of_attorney_id [String] Unique identifier of the submitted POA
+      def perform(power_of_attorney_id, form_number)
+        power_of_attorney = ClaimsApi::PowerOfAttorney.find(power_of_attorney_id)
+
+        output_path = pdf_constructor(form_number).construct(data(power_of_attorney, form_number),
+                                                             id: power_of_attorney.id)
+        upload_to_vbms(power_of_attorney, output_path)
+        ClaimsApi::PoaUpdater.perform_async(power_of_attorney.id)
+      rescue VBMS::Unknown
+        rescue_vbms_error(power_of_attorney)
+      rescue Errno::ENOENT
+        rescue_file_not_found(power_of_attorney)
+      end
+
+      def pdf_constructor(form_number)
+        if form_number == '2122A'
+          ClaimsApi::V2::PoaPdfConstructor::Individual.new
+        else
+          ClaimsApi::V2::PoaPdfConstructor::Organization.new
+        end
+      end
+
+      #
+      # Combine form_data with auth_headers and signature data
+      #
+      # @param power_of_attorney [ClaimsApi::PowerOfAttorney] Record for this poa change request
+      # @param form_number [String] Either 2122 or 2122A
+      #
+      # @return [Hash] All data to be inserted into pdf
+      def data(power_of_attorney, form_number)
+        res = power_of_attorney
+              .form_data.deep_merge({
+                                      'veteran' => {
+                                        'firstName' => power_of_attorney.auth_headers['va_eauth_firstName'],
+                                        'lastName' => power_of_attorney.auth_headers['va_eauth_lastName'],
+                                        'ssn' => power_of_attorney.auth_headers['va_eauth_pnid'],
+                                        'birthdate' => power_of_attorney.auth_headers['va_eauth_birthdate']
+                                      }
+                                    })
+
+        signatures = if form_number == '2122A'
+                       individual_signatures(power_of_attorney)
+                     else
+                       organization_signatures(power_of_attorney)
+                     end
+
+        res.merge!({ 'text_signatures' => signatures })
+        res
+      end
+
+      def organization_signatures(power_of_attorney)
+        first_name = power_of_attorney.form_data['serviceOrganization']['firstName']
+        last_name = power_of_attorney.form_data['serviceOrganization']['lastName']
+        {
+          'page2' => [
+            {
+              'signature' => "#{power_of_attorney.auth_headers['va_eauth_firstName']} " \
+                             "#{power_of_attorney.auth_headers['va_eauth_lastName']} - signed via api.va.gov",
+              'x' => 35,
+              'y' => 240
+            },
+            {
+              'signature' => "#{first_name} #{last_name} - signed via api.va.gov",
+              'x' => 35,
+              'y' => 200
+            }
+          ]
+        }
+      end
```

Comment:
Austin's pr only adds e-sign as an option for V1. This implements it for V2 in a similar way.

Reply:
oh, I was misinterpreting this as the digital signatures from the request (like v1) but I believe I see what is happening now as we're using the auth_headers for the PoA and form_data for the Veteran (pulled from MPI). And to clarify, the incoming changes for the signature in v2 would be added in Austin's tickets but you've done a bit of the work here? I'm assuming there may be some left over or do these two tickets essentially get completed with this shift? @acovrig - for awareness - are we needing additional changes? I feel like the answer is yes but just confirming.

Reply:
Assuming this pr gets merged first, I think Austin's pr would need cleaned up to remove any changes made to V1 and it would still need the registration number changes. This pr would handle how e-sign gets inserted into the pdf for 2122 and 2122a.